src/pages/index.js

```diff
@@ -119,39 +99,91 @@ function closeModal(modal) {
   modal.classList.remove("modal_opened");
 }
 
-function closeModalOnEscape(event) {
-  if (event.key === "Escape") {
+function closeModalOnEscape(evt) {
+  if (evt.key === "Escape") {
     const modals = document.querySelectorAll(".modal_opened");
     modals.forEach(closeModal);
   }
 }
 
-function handleEditProfileFormSubmit(event) {
-  event.preventDefault();
-
-  profileName.textContent = profileNameInput.value;
-  profileDescription.textContent = profileDescriptionInput.value;
-
-  closeModal(editModal);
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtnEl = evt.submitter;
+  const { avatar } = evt.target.elements;
+
+  setBtnText(saveBtnEl, false);
+  api
+    .editUserAvatar(avatar.value)
+    .then((data) => {
+      avatarEl.src = data.avatar;
+      closeModal(editAvatarModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => setBtnText(saveBtnEl, true));
 }
 
-function handleNewPostSubmit(event) {
-  event.preventDefault();
-  const link = newPostImageLinkInput.value;
-  const name = newPostNameInput.value;
-  event.target.reset();
-
-  const cardEl = getCardElement({
-    link,
-    name,
-  });
+function handleEditProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtnEl = evt.submitter;
+  const { name, about } = evt.target.elements;
+
+  setBtnText(saveBtnEl, false);
+  api
+    .editUserInfo({
+      name: name.value,
+      about: about.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => setBtnText(saveBtnEl, true));
+}
 
-  cardsList.prepend(cardEl);
-  closeModal(newPostModal);
+function handleNewPostSubmit(evt) {
+  evt.preventDefault();
+  const addPostBtnEl = evt.submitter;
+  const { name, link } = evt.target.elements;
+
+  setBtnText(addPostBtnEl, true, "Add post", "Adding post...");
+  api
+    .addPost({
+      name: name.value,
+      link: link.value,
+    })
+    .then((cardData) => {
+      const cardEl = getCardElement(cardData);
+      cardList.prepend(cardEl);
+      closeModal(newPostModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() =>
+      setBtnText(addPostBtnEl, false, "Add post", "Adding post...")
+    );
+
+  evt.target.reset();
   disableButton(postSubmitBtn, formValidationConfig);
```

Comment:
```js

  evt.target.reset();
  disableButton(postSubmitBtn, formValidationConfig);
```

Please, put it into `then` in line 158

You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

------------------------

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.